### PR TITLE
E_CLASSROOM-241 [Admin] FE Implement Admin Login

### PR DESCRIPTION
**Pull request by Teraaay**
### Issue Link

https://framgiaph.backlog.com/view/E_CLASSROOM-241

### Definition of Done
- [x] Allow Admin Login

### Related PR

BE PR Link: https://github.com/framgia/sph-classroom-els-be/pull/45

### Commands to Run

N/A

#### Main note
- Student users should not be able to log in using the login form for the admin users, [vice versa]
- Student users should not be able to access pages that are for admins only as well [vice versa]

- Sign-in button will disable if you have already submitted and will just enable if there is an error

- Use the admin account created from the seeders to log in as an admin. 
   > if you have already run the seeders command before, you should already have an admin account present in your local DB
   > to determine an Admin account it should have a user_type_id of `1`
   
#### Pages Affected
- http://localhost:3003/admin/login

### Scenarios/ Test Cases
- [ ] it should `redirect to categories list page` when `an admin user successfully logs in` 
- [ ] it should `not grant access to student accounts` when `trying to login using a student account` [vice versa]

### Screenshots
IT SHOULD LOOK LIKE THIS WHEN TRYING TO LOG IN USING A STUDENT ACCOUNT: [VICE VERSA]

image

ADMIN USERS SHOULD NOT BE ABLE TO ACCESS STUDENT PAGES [VICE VERSA]

> Admin trying to access student pages
   ADMIN TRYING TO ACCESS STUDENT PAGES

> Student trying to access admin pages
   STUDENT TRYING TO ACCESS ADMIN PAGES

### CODE REVIEW

**Comment by catampongan** [comment]
## Approved PR #83 
- [x] Able to login as an admin
- [x] Students are not allowed to access admin's login page
- [x] Admins are not allowed to access student's login page

---------
`Student trying to access and login as admin`

admin-login

`Admin trying to access as a student`

Screenshot 2022-01-31 115100

------

- Total Test Time: 0.75hr
- Total Iterations: 0
- Total Comments: 0

File: src/pages/admin/main/Login/index.module.css [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+  padding: 48px 48px;
+  border-radius: 2px;
+  height: auto;
+  width: 430px;
+}
+
+.signInText {
+  height: 45px;
+  align-self: center;
+  font-weight: 400px;
```
Comment (karlolee50) [inline src/pages/admin/main/Login/index.module.css]:
```suggestion
  font-weight: 400;
```

font-weight should not have a 'px'

File: src/pages/admin/main/Login/index.js [hunk 2]
```diff
@@ -1,7 +1,126 @@
...
+                    name="email"
+                    defaultValue=""
+                    render={({ field: { onChange, value, ref } }) => (
+                      <Form.Control
+                        onChange={onChange}
+                        value={value}
+                        ref={ref}
+                        type="email"
+                        placeholder="Enter email here"
+                        isInvalid={error === 'The password you’ve entered is incorrect.' ? '' : error}
```
Comment (karlolee50) [inline src/pages/admin/main/Login/index.js]: does the error have to be exactly 'The password..."?

is it possible to just check if the error variable has a value and make a conditional statement from there?

**Review by karlolee50 — CHANGES_REQUESTED** [review]
Review time: 20 minutes
Comment/s: 2

**Review by karlolee50 — APPROVED** [review]

**Pull request merged**